### ISSUE TITLE
Help regarding how to use Swagger API Client

### ISSUE THREAD

**Issue by serhatcetinkaya** [bitwarden/server#1144]
Hello, I have a hosted bitwarden setup running on AWS Fargate. I created client library for golang using swagger:

```
swagger-codegen generate -i https://$BITWARDEN_URL/api/specs/public/swagger.json -l go -o go-bitwarden
```

Now I want to use it from my golang application but I couldn't really understand how to use it. What I want to do is:
- authenticate using OAuth2
- do encrypted export for secrets
- export all the attachments

Can you guide me over some examples to figure our how to use it ?

**Comment by kspearrin** [comment]
The API is documented here:

https://bitwarden.com/help/article/public-api/
https://bitwarden.com/help/api/

**Comment by serhatcetinkaya** [comment]
Ok @kspearrin, I will reopen this issue because I cannot find my way out. Like I said in the original comment, I am trying to authenticate, do encrypted export and export all the attachments. 

I am struggling for days regarding how to do that. I used BURP Proxy to sniff which calls bitwarden-cli tool does and saw that when I try an encrypted export it talks to `/ciphers` endpoint, however that endpoint is not documented anywhere.

I also checked the jslib, looks like for encrypted export it is configured to use the same endpoint as well:

https://github.com/bitwarden/jslib/blob/8a3b551c6e4113035dc40a379f43f8778d208ffe/src/services/export.service.ts#L251
https://github.com/bitwarden/jslib/blob/8a3b551c6e4113035dc40a379f43f8778d208ffe/src/services/api.service.ts#L460

Is there any resource available about how to use this endpoint ? I used curl to see what it returns but it returns 401:

```
curl -s -X LIST https://$BITWARDEN_URL/api/public/ciphers -H "Authorization: Bearer ${B_TOKEN}" 
```

My token's scope is `api.organizations` . I got the client_id and client_secret from web UI (My Organization > Settings > View API Key).

Any help is appreciated
